delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphPluginModule.scala

```diff
@@ -138,10 +145,9 @@ class BlazegraphPluginModule(priority: Int) extends ModuleDef {
         cr: RemoteContextResolution @Id("aggregate"),
         ordering: JsonKeyOrdering
     ) =>
-      new BlazegraphViewsRoutes(views, viewsQuery, identities, acls, projects, progresses, coordinator)(
+      new BlazegraphViewsRoutes(views, viewsQuery, identities, acls, projects, progresses, coordinator.restart)(
```

Comment:
Your proxy error could have been caused by this: `coordinator.restart`
Since you are accessing `coordinator` here when distage might not have yet been able to get its dependencies

Reply:
It was `BlazegraphViews` that could not be proxied with a cglib error asking if it was final (and it was final).
The routes come later in the dependency tree.
With circular dependencies, we expose ourselves to this proxy problem, it also makes code and tests more complicated so getting rid of it when we can is not a bad thing.
I will see how to make the change about populating cache and view easier to understand.